### PR TITLE
PoC: Netty-based Google gRPC server using Wire for the encoding format

### PR DESCRIPTION
**Pull request by mightyguava**
This is a rough PoC of using the official Google Java gRPC implementation instead of Wire's custom implementation. There's some rough MiSK integration in `RouteGuideMiskServiceModule` to show that it can be done, but a lot more work will need to go into this to get to a production-ready state.

This PR contains working implementations of unary, client streaming, server streaming, and bi-directional streaming RPCs.

### CODE REVIEW

File: misk-grpc-vanilla/src/main/kotlin/misk/grpc/vanilla/wireserver/RouteGuideWireGrpc.kt [hunk 1]
```diff
@@ -0,0 +1,323 @@
+package misk.grpc.vanilla.wireserver
```
Comment (mightyguava) [inline misk-grpc-vanilla/src/main/kotlin/misk/grpc/vanilla/wireserver/RouteGuideWireGrpc.kt:1]: This entire file should be generated. We can either build this into (or fork) the Wire compiler, fork the Google Java compiler, or build it as a `protoc` plugin

File: misk-grpc-vanilla/build.gradle [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  compile dep.junitApi
+  compile dep.docker
+  compile dep.grpcNetty
+  compile dep.grpcProtobuf
+  compile dep.grpcStub
+  compile dep.kotlinxCoroutines
+  compile dep.wireGrpcClient
+  compile dep.wireRuntime
+  compile project(':misk')
+  compile project(':misk-testing')
```
Comment (chris-ryan-square) [inline misk-grpc-vanilla/build.gradle:66]: nit: should be in the test compile path, not normal compile

File: misk-grpc-vanilla/src/main/kotlin/misk/grpc/vanilla/wireserver/RouteGuideWireGrpc.kt (hunk 1 above)
Comment (r3mariano) [reply misk-grpc-vanilla/src/main/kotlin/misk/grpc/vanilla/wireserver/RouteGuideWireGrpc.kt:1]: I think this is the key piece

Comment (mightyguava) [reply misk-grpc-vanilla/src/main/kotlin/misk/grpc/vanilla/wireserver/RouteGuideWireGrpc.kt:1]: It is! Defining what should get generated is step 1. Generating it should be fairly easy.

Would you like to try doing it? I'm happy to get it started.

File: misk-grpc-vanilla/src/main/kotlin/misk/grpc/vanilla/wireserver/RouteGuideWireGrpc.kt [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+            .addMethod(getListFeaturesMethod())
+            .addMethod(getRecordRouteMethod())
+            .addMethod(getRouteChatMethod())
+            .build()
+          serviceDescriptor = result
+        }
+      }
+    }
+    return result
+  }
```
Comment (r3mariano) [inline misk-grpc-vanilla/src/main/kotlin/misk/grpc/vanilla/wireserver/RouteGuideWireGrpc.kt:58]: I'm curious why this is the pattern vs something like
```kotlin
  val serviceDescriptor: ServiceDescriptor
    get() { ... }
```

Comment (r3mariano) [reply misk-grpc-vanilla/src/main/kotlin/misk/grpc/vanilla/wireserver/RouteGuideWireGrpc.kt:58]: I see, it makes the thing get instantiated once?

Comment (r3mariano) [reply misk-grpc-vanilla/src/main/kotlin/misk/grpc/vanilla/wireserver/RouteGuideWireGrpc.kt:58]: It's also the what grpc uses in the codegen'd java code...

**Comment by r3mariano** [comment]
I'll close this. There's some work on this merged into https://github.com/square/wire now, though a bit unproven.

**Pull request closed without merging**